### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by svbrodersen**

### CODE REVIEW

**Comment by svbrodersen** [comment]
This those come with some of the documentation changes although they are not done. I branched off that branch, however I don't think that should be an issue.

**Review by NiklasPfister — APPROVED** [review]
Awesome thanks for the quick fixes!
If the example in the README runs now, you can merge after fixing the small comments I mentioned. Otherwise I can check tomorrow as well.

File: src/adaXT/random_forest/random_forest.py [hunk 1]
```diff
@@ -565,22 +582,38 @@ def predict_proba(self, X: np.ndarray, **kwargs) -> np.ndarray:
...
         if X is None:
             return self.__get_forest_matrix(scale=scale)
+        if self.n_jobs == 1:
+            tree_weights = []
+            for tree in self.trees:
+                tree_weights.append(
+                    predict_single_leaf(tree=tree, X=X, scale=scale))
+            return np.sum(tree_weights, axis=0) / self.n_estimators
+
+        X = shared_numpy_array(X)
```
Comment (NiklasPfister) [inline src/adaXT/random_forest/random_forest.py:616]: Can't this potentially lead to issues if X is also the shared object?

**Review by NiklasPfister — COMMENTED** [review]
I think there is still a problem but I was able to fix it if in decision_tree.pyx on line 213 in the predict_leaf_matrix function you add np.asarray() around the return value as in the other functions.

**Review by NiklasPfister — APPROVED** [review]
Nice, this fixes the problem for me.

**Pull request merged**